config/config.go

```diff
@@ -69,6 +80,137 @@ type Config struct {
 	RecoverPolicy            string        `toml:"recover_policy"`
 }
 
+type StartupParameterConfig struct {
+	Address                  string `toml:"address"`
+	LogLevel                 string `toml:"log_level"`
+	LogDir                   string `toml:"log_dir"`
+	ConfigPath               string `toml:"config_path"`
+	RootDir                  string `toml:"root_dir"`
+	CacheDir                 string `toml:"cache_dir"`
+	GCPeriod                 string `toml:"gc_period"`
+	ValidateSignature        bool   `toml:"validate_signature"`
+	PublicKeyFile            string `toml:"public_key_file"`
+	ConvertVpcRegistry       bool   `toml:"convert_vpc_registry"`
+	NydusdBinaryPath         string `toml:"nydusd_binary_path"`
+	NydusImageBinaryPath     string `toml:"nydus_image_binary_path"`
+	SharedDaemon             bool   `toml:"shared_daemon"`
+	DaemonMode               string `toml:"daemon_mode"`
+	FsDriver                 string `toml:"fs_driver"`
+	SyncRemove               bool   `toml:"sync_remove"`
+	EnableMetrics            bool   `toml:"enable_metrics"`
+	MetricsFile              string `toml:"metrics_file"`
+	EnableStargz             bool   `toml:"enable_stargz"`
+	DisableCacheManager      bool   `toml:"disable_cache_manager"`
+	LogToStdout              bool   `toml:"log_to_stdout"`
+	EnableNydusOverlayFS     bool   `toml:"enable_nydus_overlay_fs"`
+	NydusdThreadNum          int    `toml:"nydusd_thread_num"`
+	CleanupOnClose           bool   `toml:"cleanup_on_close"`
+	KubeconfigPath           string `toml:"kubeconfig_path"`
+	EnableKubeconfigKeychain bool   `toml:"enable_kubeconfig_keychain"`
+	RecoverPolicy            string `toml:"recover_policy"`
+}
+
+type SnapshotterConfig struct {
+	StartupParameterCfg StartupParameterConfig `toml:"StartupParameter"`
+}
+
+func LoadShotterConfigFile(snapshotterConfigPath string, config *SnapshotterConfig) error {
+	// get nydus-snapshotter configuration from specified path of toml file
+	if snapshotterConfigPath == "" {
+		return errors.New("snapshotter config path cannot be specified")
+	}
+	tree, err := toml.LoadFile(snapshotterConfigPath)
+	if err != nil && !(os.IsNotExist(err)) {
+		return errors.Wrapf(err, "failed to load snapshotter config file %q", snapshotterConfigPath)
+	}
+	if err := tree.Unmarshal(config); err != nil {
+		return errors.Wrapf(err, "failed to unmarshal snapshotter config file %q", snapshotterConfigPath)
+	}
+	return nil
+}
+
+func SetStartupParameter(startupParameterCfg *StartupParameterConfig, cfg *Config) error {
```

Comment:
Why not reuse `comman.Validate` ?

Reply:
If I use `comman.Validate`, I should import package.Circular reference will occur.

Reply:
Maybe move it to a separate file?